roofit/roofitcore/src/RooFactoryWSTool.cxx

```diff
@@ -2034,6 +2036,54 @@ std::string RooFactoryWSTool::SpecialsIFace::create(RooFactoryWSTool& ft, const
     // prod::name[a,b,c]
     ft.prodfunc(instName,pargs) ;
     
+  } else if (cl == "lagrangianmorph") {
+    // Perform syntax check. Warn about any meta parameters other than the ones needed
+    for (unsigned int i=1 ; i<pargv.size() ; i++) {
+      if (pargv[i].find("$fileName(")!=0 &&
+        pargv[i].find("$observableName(")!=0 &&
+        pargv[i].find("$couplings(")!=0 &&
+        pargv[i].find("$folders(")!=0 &&
+        pargv[i].find("$NewPhysics(")!=0) {
+        throw string(Form("%s::create() ERROR: unknown token %s encountered",instName, pargv[i].c_str())) ;
+      }
+    }
+ 
+    char pargsmorph[BUFFER_SIZE];
+    pargsmorph[0] = 0;
+ 
+    for (unsigned int i=0 ; i<pargv.size() ; i++) {
+      if (strlen(pargsmorph) > 0 && pargv[i].find("$NewPhysics(")!=0)
+        strlcat(pargsmorph, ",", BUFFER_SIZE);
+ 
+      else if (pargv[i].find("$NewPhysics(")==0) {
+        vector<string> subargs = ft.splitFunctionArgs(pargv[i].c_str()) ;
+        for(const auto& subarg: subargs) {
+          char buf[BUFFER_SIZE];
+          strlcpy(buf, subarg.c_str(), BUFFER_SIZE);
+          char *save;
+          char *tok = R__STRTOK_R(buf, "=", &save);
+          vector<string> parts;
+          while (tok) {
+            parts.push_back(string(tok));
+            tok = R__STRTOK_R(0, "=", &save);
+          }
+          if (parts.size() == 2){
+            ft.ws().arg(parts[0].c_str())->setAttribute("NP",atoi(parts[1].c_str()));
+          }
+          else throw string(Form("%s::create() ERROR: unknown token %s encountered, check input provided for %s",instName,subarg));
+        }
+      }
+      else {
+        vector<string> subargs = ft.splitFunctionArgs(pargv[i].c_str()) ;
+        if (subargs.size()==1){
+          string expr = ft.processExpression(subargs[0].c_str());
+          strlcat(pargsmorph, subargs[0].c_str(),BUFFER_SIZE);
+        }
+        else throw string(Form("Incorrect number of arguments in %s, have %d, expect 1",pargv[i],(Int_t)subargs.size())) ;
```

Comment:
```suggestion
        else throw string(Form("Incorrect number of arguments in %s, have %d, expect 1",pargv[i].c_str(),(Int_t)subargs.size())) ;
```
To avoid compiler warnings.